go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/ionos-cloud/cluster-api-provider-proxmox
 
-go 1.22.9
+go 1.23
```

Comment:
it's better to pin the go tool version. 

```
go 1.23.0
```

Reply:
I'll let that happen through one of the dependency update PRs. In this PR I just wanted to make the minimum set of changes for go 1.23.

Reply:
They'll just go straight for 1.23.7, anyway.

Reply:
Yeah, but will provoke some diff when someone tries to update some deps.
Cause everyone has his own go version.

Reply:
I'lll set it it to 1.23.7 because that's what the existing [go 1.23](https://github.com/ionos-cloud/cluster-api-provider-proxmox/labels/go%2F1.23) PRs will bump it to anyway.